typescript/web/src/connectors/undo-store/effects/create-label-class-and-update-label.test.tsx

```diff
@@ -1,159 +1,101 @@
-            labelClasses: [
-              {
-                id: "existing label class id",
-                name: "existing label class",
-                color: "0xaa45f7",
-              },
-            ],
-            label: {
-              id: "my label id",
-              labelClass: {
-                id: "previous label class id",
-              },
-            },
-          },
-        };
-      }),
-    },
-  };
-});
+import { createCreateLabelClassAndUpdateLabelEffect } from "./create-label-class-and-update-label";
+import {
+  APOLLO_MOCKS,
+  CREATE_LABEL_CLASS_ACTION_MOCK,
+  DELETE_LABEL_CLASS_ACTION_MOCK,
+  UPDATE_LABEL_CLASS_ACTION_MOCK,
+} from "./label-and-label-class.fixtures";
 
 const { perform } = useUndoStore.getState();
+const [PREVIOUS_LABEL_CLASS_DATA, NEW_LABEL_CLASS_DATA] =
+  DEEP_DATASET_WITH_CLASSES_DATA.labelClasses;
 
 describe("CreateLabelClassAndUpdateLabelEffect", () => {
+  let mockClient: ApolloClientWithMockLink;
+
   beforeEach(async () => {
     jest.clearAllMocks();
+    mockClient = getApolloMockClient(APOLLO_MOCKS);
     useLabelingStore.setState({
-      selectedLabelClassId: "previous label class id",
+      selectedLabelClassId: PREVIOUS_LABEL_CLASS_DATA.id,
     });
     await perform(
       createCreateLabelClassAndUpdateLabelEffect(
         {
-          name: "new label class",
-          color: "0xaa45f8",
-          datasetId: "a dataset id",
-          selectedLabelId: "my label id",
+          name: NEW_LABEL_CLASS_DATA.name,
+          color: NEW_LABEL_CLASS_DATA.color,
+          datasetId: DEEP_DATASET_WITH_CLASSES_DATA.id,
+          selectedLabelId: BASIC_LABEL_DATA.id,
         },
-        { client }
+        { client: mockClient.client }
       )
     );
   });
 
   it("creates the label class and update the label and the labeling store", async () => {
-    expect(client.mutate).toHaveBeenNthCalledWith(
-      1,
-      expect.objectContaining({
-        variables: {
-          data: expect.objectContaining({
-            name: "new label class",
-            color: "0xaa45f8",
-            datasetId: "a dataset id",
-          }),
-        },
-      })
-    );
-
-    // The id is generated by the effect. We retrieve it from the first mutation
-    // which is "createLabelClass"
-    const { id } = (client.mutate as jest.Mock).mock.calls[0][0].variables.data;
-
-    expect(client.mutate).toHaveBeenNthCalledWith(
-      2,
-      expect.objectContaining({
-        variables: {
-          where: { id: "my label id" },
-          data: { labelClassId: id },
-        },
-      })
-    );
+    expect(CREATE_LABEL_CLASS_ACTION_MOCK.result).toHaveBeenCalledWith({
+      data: expect.objectContaining({
+        name: NEW_LABEL_CLASS_DATA.name,
+        color: NEW_LABEL_CLASS_DATA.color,
+        datasetId: DEEP_DATASET_WITH_CLASSES_DATA.id,
+      }),
+    });
+    const newLabelClassId = (CREATE_LABEL_CLASS_ACTION_MOCK.result as jest.Mock)
+      .mock.calls[0][0].data.id;
```

Comment:
As you've previously done in another PR, the result `jest.fn` declaration should be extracted in a dedicated const in the fixture, so that you can directly reuse it and avoid the need to cast the `result` as `jest.Mock`.

i.e.:

```suggestion
    const [{data: {id: newLabelClassId]}] = createLabelClassActionMockResult.mock.calls[0];
```
(_Maybe it's not extractable in a one-liner because of nullable types but you get the idea_)